### ISSUE TITLE
Add optional path to qualify source links if not a root of git repo

### ISSUE THREAD

**Issue by grahamcarlyle** [borkdude/quickdoc#26]
When generating docs from a location other than the root of the github repo, the source links do not work.

Adding a `:git/path` option could provide this extra path information to generate the correct source links.

**Comment by grahamcarlyle** [comment]
I added a PR for this https://github.com/borkdude/quickdoc/pull/27

**Comment by borkdude** [comment]
It's not immediately clear to me how I would use this and when. You could solve this already using`:filename-fn` right, but this is to make it more EDN-compatible?

Is it possible to add an example to the README.md?

I really need to write some tests as well for this project. An example would help me to write that test as well.

**Comment by grahamcarlyle** [comment]
The need I have is to generate docs when the source-path specified to quickdoc isn't relative to the root of the repository. I have multiple projects within the same repository and want to generate docs for them, but the default source link generated assumes the source path is relative to the repo root, rather than relative to a project sub-dir.

I can solve the problem already using `:filename-fn` but I think a simpler existing solution for a deps.edn project is to use a custom `:source-uri`

e.g. 
```
:exec-args {
  :source-uri "https://github.com/borkdude/quickdoc/blob/{branch}{path}/{filename}#L{row}-L{end-row}"
  :git/branch "master"
  :git/path "/my-project"
  :source-paths ["src/main/clojure"]}
```
The PR I've sent is just making that solution slightly simpler by adjusting the default github template and the quickdoc options to accept a `path` variable, and also to make the template use the `:github/repo` variable.

I've updated the PR with an example in the README, and some tests for var-source. I also switched to using an absolute path rather than a relative one as on reflection that seemed better.

I think the name `:git/path` isn't great but hopefully the intent is clear.

**Comment by borkdude** [comment]
So after reading the tests, what this feature is, is "add a prefix to the filename", right?

We could call this option `:filename-add-prefix` to _add_ something in front of the filename, perhaps?

Additionally, we could also have an option `:filename-remove-prefix` to _remove_ something from the front of the filename, which is also a use case I've had (which why `:filename-fn` was introduced).

**Comment by grahamcarlyle** [comment]
Yes that sounds good to me.

**Comment by borkdude** [comment]
OK, do you want to make this change or want me to do it?

**Comment by grahamcarlyle** [comment]
I've updated the PR with these named options, but happy for you to either ignore or use this as a basis to make the change. 

I removed the entry for :git/path in the README but wasn't sure whether adding an equivalent one for the add and remove options was going to be useful. I also wasn't sure whether the add, remove and filename-fn options should be exclusive or could work alongside each other, so went with exclusive.
